packages/dialtone-vue2/recipes/conversation_view/wysiwyg_editor/wysiwyg_editor.vue

```diff
@@ -0,0 +1,574 @@
+        size="sm"
+        importance="clear"
+        kind="inverted"
+        @click="onItalicTextToggle"
+      >
+        <template #icon>
+          <dt-icon
+            name="italic"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+
+      <dt-button
+        v-if="showUnderlineButton"
+        data-qa="dt-wysiwyg-editor-underline-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        :class="{ 'd-bgc-black-300 d-fc-black-900': $refs.richTextEditor?.editor?.isActive('underline') }"
+        size="sm"
+        importance="clear"
+        kind="inverted"
+        @click="onUnderlineTextToggle"
+      >
+        <template #icon>
+          <dt-icon
+            name="underline"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+
+      <dt-button
+        v-if="showStrikeButton"
+        data-qa="dt-wysiwyg-editor-strike-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        :class="{ 'd-bgc-black-300 d-fc-black-900': $refs.richTextEditor?.editor?.isActive('strike') }"
+        size="sm"
+        importance="clear"
+        kind="inverted"
+        @click="onStrikethroughTextToggle"
+      >
+        <template #icon>
+          <dt-icon
+            name="strikethrough"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+
+      <dt-button
+        v-if="showListItemsButton"
+        data-qa="dt-wysiwyg-editor-list-items-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        :class="{ 'd-bgc-black-300 d-fc-black-900': $refs.richTextEditor?.editor.isActive('bulletList') }"
+        size="sm"
+        importance="clear"
+        kind="inverted"
+        @click="onBulletListToggle"
+      >
+        <template #icon>
+          <dt-icon
+            name="list-bullet"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+
+      <dt-button
+        v-if="showAddLink.showAddLinkButton"
+        data-qa="dt-wysiwyg-editor-add-link-btn"
+        class="h:d-bgc-black-300 d-fc-black-700 d-mx4"
+        size="sm"
+        importance="clear"
+        kind="inverted"
+        @click="openLinkInputModal"
+      >
+        <template #icon>
+          <dt-icon
+            name="link-2"
+            size="200"
+            class="d-fw-bold"
+          />
+        </template>
+      </dt-button>
+    </div>
+
+    <!-- Add/Remove link modal -->
+    <dt-modal
+      :show="showLinkInput"
+      :title="showAddLink.setLinkModalTitle"
+      :visually-hidden-close="true"
+      :visually-hidden-close-label="'Close link input modal'"
+      data-qa="dt-wysiwyg-editor-link-input-modal"
+      show-footer
+      kind="default"
+      size="default"
+      :close-button-props="{ ariaLabel: 'Close the dialog' }"
+      @update:show="(s) => updateInputModal(s)"
+    >
```

Comment:
Pressing enter while within the modal should confirm.

Reply:
Bit of a weird bug here.. If I click within the modal (not on a button or anything) it will actually set focus on the rich text input outside of the modal.

Reply:
Yes I've been trying to fix this I think it has to do with the fact that some event is propagated and handled to set focus on the editor. Already tried to prevent it by using @click.native.stop but evidently something else is slipping through. I'll take another look and see if I can find what's causing it.